File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// Functions:
int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
void print_grade(float i);

// This program takes a text and determinates its reading level.
int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float subindex = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(subindex);

    print_grade(index);
}

int count_letters(string s)
{
    int lenght = strlen(s);
    int ls;
    int count = 0;
    for(int i = 0; i < lenght; i++)
    {
        ls = (int) tolower(s[i]);
        if(ls >= 97 && ls <= 122)
        {
            count++;
        }
    }
    return count;
}

int count_words(string s)
{
    int lenght = strlen(s);
    int count = 1;
    for(int i = 0; i < lenght; i++)
    {
        if(isspace(s[i]))
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string s)
{
    int lenght = strlen(s);
    int count = 0;
    for(int i = 0; i < lenght; i++)
    {
        if((s[i] == '.') || (s[i] == '!') || (s[i] == '?'))
        {
            count++;
        }
    }
    return count;
}

void print_grade(float i)
{
    if(i >= 16)
    {
        printf("Grade 16+\n");
    }
    else if(i < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(i));
    }
}

File: Week_6/sentimental-readability/readability.py
from cs50 import get_string


# This program takes a text and determinates its reading level.
def main():

    # promt the user for a Text
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # calculate the reading level
    l = letters / words * 100
    s = sentences / words * 100
    index = 0.0588 * l - 0.29 * s - 15.8
    index = round(index)

    # print the reading level
    print_grade(index)


def count_letters(s):
    count = 0

    # loop through the string and count the letters
    for char in s:
        if char.isalpha():
            count += 1
    return count

def count_words(s):
    count = 1

    # loop through the string and count the words
    for char in s:
        if char == " ":
            count += 1
    return count


def count_sentences(s):
    count = 0

    # loop through the string and count the sentences
    for char in s:
        if char == "." or char == "!" or char == "?":
            count += 1
    return count


def print_grade(i):

    if i >= 16:
        print(f"Grade 16+")
    elif i < 1:
        print(f"Before Grade 1")
    else:
        print(f"Grade {i}")


main()